graphene/types/typemap.py

```diff
@@ -119,6 +119,8 @@ def construct_enum(cls, map, type):
     @classmethod
     def construct_objecttype(cls, map, type):
         from .definitions import GrapheneObjectType
+        if type._meta.name in map:
```

Comment:
For consistency we should also check if `map[type._meta.name].graphene_type == type` :)